cli/src/compile.rs

```diff
@@ -258,3 +261,31 @@ pub fn compile_files(
     println!("{output}");
     Ok(output)
 }
+
+/// Compiles target file path.
+///
+/// If `silent` no solc related output will be emitted to stdout.
+///
+/// If `verify` and it's a standalone script, throw error. Only allowed for projects.
+pub fn compile_target(
+    target_path: &PathBuf,
```

Comment:
requiring `&Path` is preferred so you don't need an allocated path, same with `&str` instead of `&String`

```suggestion
    target_path: &Path,
```